frontend/public/components/utils/volume-icon.jsx

```diff
@@ -1,17 +1,26 @@
 import * as _ from 'lodash-es';
 import * as React from 'react';
 import {VolumeSource} from '../../module/k8s/pods';
+import { ResourceLink } from './resource-link';
 
-export const VolumeIcon = ({kind}) => {
+export const VolumeLink = ({kind, name, namespace}) => {
   const faClasses = _.fromPairs([
     [VolumeSource.emptyDir.id, 'fa-folder-open-o'],
     [VolumeSource.hostPath.id, 'fa-files-o'],
-    [VolumeSource.secret.id, 'fa-lock'],
   ]);
   const faClass = faClasses[kind];
+  const link = _.get(VolumeSource[kind], 'link', false);
```

Comment:
We're calling this `link` but really it's a k8s kind? We should give this a better name. Maybe `k8sKind` and `volumeSourceKind` for the other variable to avoid confusion.